### ISSUE TITLE
Blueprint types are overly strict

### ISSUE THREAD

**Issue by rherms** [palantir/blueprint#1813]
Some blueprint types that require strings actually work with JSX.Element as well.
This mostly include the text for certain components such as "cancelButtonText" in Alert or "text" in Button. For example, I can pass in a <span> to these and cast it to "any" and it will render correctly. It would be nice to not have to cast these to "any".

**Comment by llorca** [comment]
`Alert` is explicitly opinionated and restrictive. If you need more flexibility, use `Dialog`.

We've never had issues with `text` prop in `Button` being a string as we allow other button aspects to be customized through the other props.

Gonna close this out, though feel free to reopen if you have a real use case to share regarding button text.

**Comment by rherms** [comment]
@llorca The components all work fine when passing in elements such as <span>s into them as props. However, the types for typescript do not allow this. Why is that? Instead of making these props strings, shouldn't they be something like `string | JSX.Element`?

**Comment by cmslewis** [comment]
@rherms types are a means of restricting "bad" usage that doesn't conform to the design system. 

e.g. in a `MenuItem`, `text` must currently be a `string`, because allowing `span`s (`JSX.Element`s) would give people a way to bold, italicize, and underline text in menu items, which we didn't want to support for lack of a clear use case when we designed the component.

As @llorca mentioned, `Alert`s and `Button`s are similarly restrictive - the typings confine you to good practice based on our understanding of the use case when we designed the component.

If we see clear use cases that argue for broader type support, we can discuss changing the types. For your use case, what if you just unpack the text from your `span`s and add any relevant `className`s to the parent components? Would that be sufficient for you?
